### ISSUE TITLE
Add Export Annotation for Source Code

### ISSUE THREAD

**Issue by iiMidknightii** [godotengine/godot-proposals#9749]
### Describe the project you are working on

Any project that uses dynamic code to create custom shaders, nodes, resources, etc.

### Describe the problem or limitation you are having in your project

Sometimes, when working with instances of nodes or resources, I'd like to be able to export a property that can customize create custom dynamic code.  This can happen in several different circumstances like states in a state machine, status effects with custom processing, custom shaders, etc.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add an annotation, e.g. `@export_source` to create a `String` variable with syntax highlighting in the editor.  The language for the source code editor could default to the language used in the script that exported the variable, but maybe it could also be possible to export source code of another language using a passed in string, e.g. `@export_source("GDScript")`, `@export_source("GDShader")`, etc.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![Screenshot 2024-05-15 195043](https://github.com/godotengine/godot-proposals/assets/89699177/1997105a-499f-41b1-b0a6-1d019b3a3a4e)

The annotation would look functionally the same as `@export_multiline`, except any code written in it would be highlighted according to the language specified.  For instance, in the screenshot above, the code is for a GDScript based node, so the text box becomes a code editor that highlights according to GDScript convention.  I'm not sure if it needs to do the full warning/error processing since it may be desirable to put code words or replacement statements into the source for the exporting script to later replace.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Technically, `@export_multiline` already does this, just without the syntax highlighting.

### Is there a reason why this should be core and not an add-on in the asset library?

I'm not sure if annotations can be added via add-ons.

**Comment by dalexeev** [comment]
> If it would be enough for you to just highlight the code in the inspector, then I think that this could be implemented as a plugin, without annotation.

See [dalexeev/godot-inspector-code-highlighter](https://github.com/dalexeev/godot-inspector-code-highlighter).

**Comment by iiMidknightii** [comment]
> Personally, editing code in a small inspector field looks inconvenient.

True, but there is already a "maximize" button next to multiline text fields that expands it into a separate window.  I think if the code was more complex than just simple logic, a user would do that.
